### PR TITLE
closes #368

### LINKED ISSUE #368: `connected_components` dispatch too broad?
`connected_components` doesn't seem to have a meaningful return when called on a directed graph.

``` julia
julia> g = DiGraph(5)
{5, 0} directed graph

julia> [add_edge!(g, Pair(x...)) for x in Array[[3,2],[3,4],[4,5],[5,1]]]
4-element Array{Pair{Int64,Int64},1}:
 edge 3 - 2
 edge 3 - 4
 edge 4 - 5
 edge 5 - 1

julia> connected_components(g)
3-element Array{Array{Int64,1},1}:
 [1]    
 [2]    
 [3,4,5]

julia> g2 = DiGraph(5)
{5, 0} directed graph

julia> [add_edge!(g2, Pair(x...)) for x in Array[[1,2],[1,3],[3,4],[3,5]]]
4-element Array{Pair{Int64,Int64},1}:
 edge 1 - 2
 edge 1 - 3
 edge 3 - 4
 edge 3 - 5

julia> connected_components(g2)
1-element Array{Array{Int64,1},1}:
 [1,2,3,4,5]

julia> symdiff(strongly_connected_components(g), strongly_connected_components(g2))
0-element Array{Array{Int64,1},1}

julia> symdiff(weakly_connected_components(g), weakly_connected_components(g2))
0-element Array{Array{Int64,1},1}
```

Should it's dispatch be limited to undirected graphs where it has meaning? Or could the documentation change of 9873d0f6274e584359 be reverted to specify that's the type it works on?

edit: This is on julia 0.4.5 using LightGraphs master.

### PR DESCRIPTION
**Pull request by sbromberger**
`is_connected` and `connected_components` now throw method errors for directed graphs.

### CODE REVIEW

File: src/connectivity.jl [hunk 1]
```diff
@@ -238,25 +247,13 @@ function condensation end
...
 """
     attracting_components(g)
 
-Return the condensation graph associated with `g`.
-
-The condensation `h` of a graph `g` is the directed graph where every node
-in `h` represents a strongly connected component in `g`, and the presence
-of an edge between between vertices in `h` indicates that there is at least one
-edge between the associated strongly connected components in `g`. The node
-numbering in `h` corresponds to the ordering of the components output from
```
Comment (jpfairbanks) [inline src/connectivity.jl]: why are these docs deleted?

Comment (sbromberger) [inline src/connectivity.jl]: because those are docs for `condensation`, not `attracting components`.

**Comment by sbromberger** [comment]
@jpfairbanks - mind giving this a review as well? Thanks.

**Comment by sbromberger** [comment]
I think this is the last PR holding up tagging of 0.8.

**Comment by sbromberger** [comment]
I plan on merging this tomorrow by end of day. It's a well-documented minor change that can be easily reversed if it causes problems. I'd like to tag 0.8.0 by next week.

**Pull request merged**